### ISSUE TITLE
some kde connect functions don't work on GSconnect

### ISSUE THREAD

**Issue by paulo8448** [andyholmes/gnome-shell-extension-gsconnect#878]
> ***ATTENTION**: GSConnect only supports the latest, stable version of GNOME. We are no longer accepting bug reports for previous versions.*

**Describe the bug**

the most important functions of kde connect that is about controlling the computer and the slide show function are not working at first I thought it would be a wayland problem but in xorg the same problem is happening

I use gnome manjaro
version 3.36
latest version of gsconnect

**Comment by ferdnyc** [comment]
Hmm, confirmed. Same here.

@andyholmes :
Running what I _believe_ is stock v38 under GNOME Shell 3.36.3 (current Fedora 32 packages), and generating a Support Log, I see that `input.js` is failing to set up its RemoteDesktop session with Mutter, the first time it receives an "actionable" input packet. It logs "`this._session is null`" from `pressKey()`, and that's the last sign of life from any of the input-processing code. All further mouse/keyboard messages receive only deafening silence in response. 

Support log contents, with the superfluous indentation trimmed off:
```

GSConnect Version: 38
GSConnect Install: user
GJS: 16403
XDG_SESSION_TYPE: x11
GDMSESSION: gnome
--------------------------------------------------------------------------------
-- Logs begin at Mon 2020-06-08 01:57:01 EDT, end at Thu 2020-06-18 09:36:14 EDT. --
[/service/protocol/lan.js:_onIdentity:315]: {
                                "id": 1592487362922,
                                "type": "kdeconnect.identity",
                                "body": {
                                  "deviceId": "1a93f4cf61e7ee6d",
                                  "deviceName": "Galaxy Note8",
                                  "protocolVersion": 7,
                                  "deviceType": "phone",
                                  "incomingCapabilities": [
                                    "kdeconnect.telephony.request_mute",
                                    "kdeconnect.notification",
                                    "kdeconnect.ping",
                                    "kdeconnect.notification.reply",
                                    "kdeconnect.notification.action",
                                    "kdeconnect.share.request",
                                    "kdeconnect.clipboard.connect",
                                    "kdeconnect.runcommand",
                                    "kdeconnect.contacts.request_all_uids_timestamps",
                                    "kdeconnect.sms.request_conversations",
                                    "kdeconnect.telephony.request",
                                    "kdeconnect.mpris",
                                    "kdeconnect.sms.request_conversation",
                                    "kdeconnect.findmyphone.request",
                                    "kdeconnect.systemvolume",
                                    "kdeconnect.mousepad.keyboardstate",
                                    "kdeconnect.sftp.request",
                                    "kdeconnect.share.request.update",
                                    "kdeconnect.notification.request",
                                    "kdeconnect.mousepad.request",
                                    "kdeconnect.photo.request",
                                    "kdeconnect.sms.request",
                                    "kdeconnect.contacts.request_vcards_by_uid",
                                    "kdeconnect.mpris.request",
                                    "kdeconnect.battery.request",
                                    "kdeconnect.clipboard"
                                  ],
                                  "outgoingCapabilities": [
                                    "kdeconnect.sms.messages",
                                    "kdeconnect.telephony",
                                    "kdeconnect.mpris",
                                    "kdeconnect.notification",
                                    "kdeconnect.contacts.response_uids_timestamps",
                                    "kdeconnect.findmyphone.request",
                                    "kdeconnect.ping",
                                    "kdeconnect.mousepad.keyboardstate",
                                    "kdeconnect.share.request",
                                    "kdeconnect.contacts.response_vcards",
                                    "kdeconnect.notification.request",
                                    "kdeconnect.mousepad.echo",
                                    "kdeconnect.clipboard.connect",
                                    "kdeconnect.mousepad.request",
                                    "kdeconnect.presenter",
                                    "kdeconnect.sftp",
                                    "kdeconnect.photo",
                                    "kdeconnect.runcommand.request",
                                    "kdeconnect.mpris.request",
                                    "kdeconnect.systemvolume.request",
                                    "kdeconnect.battery",
                                    "kdeconnect.clipboard"
                                  ],
                                  "tcpPort": 1716,
                                  "tcpHost": "192.168.12.102"
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487364842,
                                "type": "kdeconnect.mpris",
                                "body": {
                                  "playerList": []
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487364843,
                                "type": "kdeconnect.mpris",
                                "body": {
                                  "playerList": []
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487366517,
                                "type": "kdeconnect.mousepad.request",
                                "body": {
                                  "specialKey": 8
                                }
                              }
[/service/components/input.js:_ensureAdapter:414]: "Creating Mutter RemoteDesktop session"
[/service/components/input.js:pressKey:544]: this._session is null
pressKey@/home/ferd/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/components/input.js:544:13
_handleInput@/home/ferd/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/plugins/mousepad.js:194:33
handlePacket@/home/ferd/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/plugins/mousepad.js:145:26
receivePacket@/home/ferd/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:441:29
receive/<@/home/ferd/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/protocol/core.js:299:32
@/home/ferd/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:1133:17
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487366734,
                                "type": "kdeconnect.mousepad.request",
                                "body": {
                                  "specialKey": 8
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487367005,
                                "type": "kdeconnect.mousepad.request",
                                "body": {
                                  "specialKey": 8
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487367483,
                                "type": "kdeconnect.mousepad.request",
                                "body": {
                                  "specialKey": 9
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487367661,
                                "type": "kdeconnect.mousepad.request",
                                "body": {
                                  "specialKey": 9
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487367775,
                                "type": "kdeconnect.mousepad.request",
                                "body": {
                                  "specialKey": 9
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487368634,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0027488707564771175,
                                  "dy": 0.005296783521771431
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487368653,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0027488707564771175,
                                  "dy": 0.005296783521771431
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487368672,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0027488707564771175,
                                  "dy": 0.005296783521771431
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487368690,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "stop": true
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369036,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0027488707564771175,
                                  "dy": 0.005296783521771431
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369055,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0027488707564771175,
                                  "dy": 0.005296783521771431
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369073,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0027488707564771175,
                                  "dy": 0.005296783521771431
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369092,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": -0.0011732482817023993,
                                  "dy": -0.0037380980793386698
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369111,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": -0.0012464903993532062,
                                  "dy": -0.00615692138671875
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369131,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": -0.00016525268438272178,
                                  "dy": -0.0047639464028179646
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369150,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0003479003789834678,
                                  "dy": -0.0017037963261827826
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369168,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0006225585821084678,
                                  "dy": -0.0031883239280432463
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369187,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0015206908574327826,
                                  "dy": -0.00624847412109375
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369207,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0008610534714534879,
                                  "dy": -0.005039062350988388
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369226,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0007695007370784879,
                                  "dy": 0.0006605529924854636
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369245,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0011540221748873591,
                                  "dy": 0.0025479125324636698
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369264,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0004760742012877017,
                                  "dy": 0.0005506896995939314
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369283,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": -0.000824890099465847,
                                  "dy": 0.00007415770960506052
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369302,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": -0.0004953002789989114,
                                  "dy": 0.0015032958472147584
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369322,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": -0.000824890099465847,
                                  "dy": 0.0033906553871929646
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369341,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": -0.0004216003289911896,
                                  "dy": 0.0027127075009047985
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369359,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0002563476446084678,
                                  "dy": 0.00038543701521120965
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369379,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": -0.001007995568215847,
                                  "dy": -0.00036575316335074604
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369397,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": -0.0016859435709193349,
                                  "dy": 0.0027859497349709272
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369417,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": -0.0009901428129523993,
                                  "dy": 0.005260162055492401
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369438,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0006042480235919356,
                                  "dy": 0.001979827880859375
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369455,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0016305542085319757,
                                  "dy": -0.0010620116954669356
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369473,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0014291381230577826,
                                  "dy": -0.0015568542294204235
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369493,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0008793639717623591,
                                  "dy": 0.0011549376649782062
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369512,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.001374206505715847,
                                  "dy": 0.0055718994699418545
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369532,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.002473754808306694,
                                  "dy": 0.007917022332549095
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369550,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0022901915945112705,
                                  "dy": 0.007055968977510929
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369569,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": -0.0006784057477489114,
                                  "dy": 0.0030610656831413507
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369589,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": -0.0036470030900090933,
                                  "dy": -0.0029864502139389515
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369607,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": -0.0014845276018604636,
                                  "dy": -0.0039028930477797985
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369627,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0032620239071547985,
                                  "dy": -0.0017037963261827826
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369648,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0034634398762136698,
                                  "dy": -0.006853180006146431
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369665,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.0010807800572365522,
                                  "dy": -0.013579101301729679
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369684,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": -0.002858733991160989,
                                  "dy": -0.014055175706744194
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369703,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": -0.006963958498090506,
                                  "dy": -0.0070729064755141735
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369722,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": -0.005589294247329235,
                                  "dy": 0.0027676390018314123
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369741,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": -0.000641784630715847,
                                  "dy": 0.009035339578986168
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369761,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.005497283767908812,
                                  "dy": 0.009713287465274334
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369781,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.009180908091366291,
                                  "dy": 0.011380920186638832
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369800,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "dx": 0.006413726601749659,
                                  "dy": 0.009456481784582138
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487369821,
                                "type": "kdeconnect.presenter",
                                "body": {
                                  "stop": true
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487371415,
                                "type": "kdeconnect.mousepad.request",
                                "body": {
                                  "specialKey": 25
                                }
                              }
[/service/protocol/core.js:receive/<:298]: Galaxy Note8: {
                                "id": 1592487372917,
                                "type": "kdeconnect.mousepad.request",
                                "body": {
                                  "specialKey": 14
                                }
                              }
```

**Comment by ferdnyc** [comment]
(I looked around for any other useful logs, in `dmesg`, in `$HOME/.local/share/xorg/Xorg.1.log`, in the system journal... Came up with zilch.)

**Comment by andyholmes** [comment]
I think this is probably a duplicate of #869, should be fixed in c5f42f5. Feel free to re-open if you can reproduce at that commit.

**Comment by ferdnyc** [comment]
It certainly looks identical, yeah. I'll give the fix a spin later.
